.github/workflows/cross-version-tests.yml

```
@@ -27,12 +28,9 @@ on:
   schedule:
     # Run this workflow daily at 13:00 UTC
     - cron: "0 13 * * *"
-  pull_request:
-    branches:
-      - master
 
 concurrency:
-  group: ${{ github.workflow }}-${{ github.ref == 'refs/heads/master' && github.run_number || github.ref }}
+  group: ${{ github.workflow }}-${{ github.run_number || github.ref }}
```

```suggestion
  group: ${{ github.workflow }}-${{ github.head_ref || github.run_id }}
```

I don't think `${{ github.workflow }}-${{ github.run_number || github.ref }}` cancel any jobs because `github.run_number`. We can use `${{ github.workflow }}-${{ github.head_ref || github.run_id }}` (ref: 
Ref: https://docs.github.com/en/actions/using-jobs/using-concurrency#example-using-a-fallback-value).